src/index.ts

```diff
@@ -52,15 +56,17 @@ export async function initializeFromJsonFile(jsonFilepath: string, debug?: boole
  * @param aiKey Key of Application Insights.
  * @param debug If set as true, debug information be printed to console.
  */
-export function initialize(extensionId: string, version: string, aiKey: string, debug?: boolean): void {
+export function initialize(extensionId: string, version: string, aiKey: string, options?: IOptions): void {
     if (reporter) {
         throw new Error("TelemetryReporter already initialized.");
     }
 
     if (aiKey) {
         reporter = new TelemetryReporter(extensionId, version, aiKey);
     }
-    isDebug = !!debug;
+    // @ts-ignore
+    reporter.firstParty = !!(options && options.firstParty);
```

Comment:
Do this mean each extension itself should detect whether it's vsonline, and then set firstParty?

Reply:
Seems the upstream lib handles this: https://github.com/microsoft/vscode-extension-telemetry/blob/14f14f4cbfe962c5630ba4d33928b88c4314ff10/src/telemetryReporter.ts#L139-L144

Reply:
so it should be `reporter.firstParty = report.firstParty || !!(options && options.firstParty);`?

If the upstream already marks it as firstParty, we shouldn't override it, right?

Reply:
Or only if options.firstParty is not empty (undefined or null), then override reporter.firstParty.

Reply:
No, it's a hack. Upstream set `firstParty` to true only when the AI key starts with "AIF-", but ours are not.  In our cases, e.g. debubger, testrunner, maven, `reporter.firstParty` is false when initialized.

Reply:
The upstream check logic is running at the TelemetryReporter constructor. For example, if it's vscode-java-pack, the upstream will marks it as firstParty during `new TelemetryReporter()`. Your change `reporter.firstParty = !!(options && options.firstParty)` is running after that, and it will override firstParty to false if java pack didn't provide `options` explicitly.

Reply:
> Your change reporter.firstParty = !!(options && options.firstParty) is running after that, and it will override firstParty to false if java pack didn't provide options explicitly.

Yes of course. All **our** extensions should **explicitly** provide `options` and declare to be "firstParty".  The wrapper should not decide on behalf of the end extensions... (remember it's a library, and maybe used by others..).

Reply:
i know it's a library. But it should not change the default behavior of the upstream library. 

Of course, if whatever the user explicitly sets via options, then you could regard the user's config, and override it. But the options (options?: IOptions) is optional, if the user didn't set anything, Shouldn't you follow the default library behavior?

Reply:
Yes, you are right, for the case that an extension using "AIF-" key but not setting the options... I think we just misunderstood each other by talking about different cases. 

I'll change it as below:
```ts
if (options && options.firstParty) {
  reporter.firstParty = options.firstParty;
}
```

Is it ok with you?

Reply:
looks good.